firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/settings/SettingsController.java

```diff
@@ -212,39 +211,33 @@ public Settings getSettings() {
                   // Update the regular settings.
                   settings.set(fetchedSettings);
 
-                  // Signal the app settings on any Tasks that already exist, and then replace the
-                  // task so
-                  // that any new callers get the new app settings instead of any old ones.
-                  appSettingsData.get().trySetResult(fetchedSettings.getAppSettingsData());
-                  TaskCompletionSource<AppSettingsData> fetchedAppSettings =
-                      new TaskCompletionSource<>();
-                  fetchedAppSettings.trySetResult(fetchedSettings.getAppSettingsData());
-                  appSettingsData.set(fetchedAppSettings);
```

Comment:
Did you verify that we don't need to do this?

Reply:
It works as expected in the happy path. I found a bug that can occur when there's no network access, but it predates this PR by at least several versions. So I'm fairly confident the above lines are fine, but I'll work it all out when I fix that bug.